README.md

```diff
@@ -1,3 +1,19 @@
 # CoreDNS
 
-CoreDNS is a DNS server.
+CoreDNS is a DNS  server. And this is a Kubernetes Charm to deploy CoreDNS on Kubernetes with Juju.
+
+## How to build the charm
+
+Make sure you've got the [coredns-interface](https://github.com/DomFleischmann/coredns-interface) in your $CHARM_INTERFACES_DIR and simply build the charm with:
+``` charm build charm-coredns -o ..```
+
+## How to deploy the charm
+
+For deploying the charm you will have to have a kubernetes cluster bootstrapped with juju to be able to deploy kubernetes charms. Than you will simply need to execute the following in the charm builds dir:  
+``` juju deploy ./coredns --resource coredns-image=rocks.canonical.com:443/cdk/coredns/coredns-amd64:1.6.2 ```
+
+## How to add the relation
+You can use the coredns interface to automatically configure the CoreDNS charm with Charmed Kubernetes. For that execute the following inside of the Charmed Kubernetes model:
```

Comment:
"automatically configure" - let's add a little detail about what happens when this relation is added (e.g. kubelet is reconfigured with the correct dns server ip address, or something similar).